izumi-reflect/izumi-reflect/src/test/scala_2/izumi/reflect/test/TagTest.scala

```diff
@@ -537,6 +537,21 @@ class TagTest extends AnyWordSpec with XY[String] {
       assert(zy.tagA.isSuccess)
     }
 
+    "resolve type members up to defining class" in {
+      trait A {
+        class X
+        val xa = Tag[X]
+      }
+
+      trait B extends A {
+        val xb = Tag[X]
+      }
+
+      object B extends B
+
+      assert(B.xa.tag == B.xb.tag)
```

Comment:
Could you also add a check that `Tag[A#X].tag == B.(xa/xb).tag)`?

Reply:
Sure! Done